src/legacy/core_plugins/embeddable_api/public/panel/panel_header/panel_actions/add_panel/add_panel_flyout.test.tsx

```diff
@@ -79,7 +81,9 @@ test('create new calls factory.adds a panel to the container', async done => {
 
   await nextTick();
 
-  (getModalContents().props as ContactCardInitializerProps).onCreate({
+  const overlayMock = npStart.core.overlays as jest.Mocked<PublicMethodsOf<OverlayStart>>;
```

Comment:
I liked the previous solution with `import { coreStartMock } from '../../../../../../../../core/public/ui_new_platform.test.mocks';` more
- you don't have to type case manually.
- jest.mock('ui/new_platform') requires 2 imports now.
- in other places you still have to do `jest.mock('ui/new_platform')`.

Reply:
yeah, having to find the right type to cast your mock into is quite tricky and not very user friendly if you're not familiar with all the core internals, e.g. it's not obvious you should be casting to `PublicMethodsOf<OverlayStart>` instead of `OverlayStart`.

@joshdover What do you think of the two options? One of the disadvantages of the previous approach was the insanely long relative path names, but I should be able to use absolute paths which will make it much more readable i.e. `import { coreStartMock } from 'src/core/public/ui_new_platform.test.mocks'`